docs/advanced/document-store/redis.md

```diff
@@ -0,0 +1,334 @@
+
+```{caution}
+Currently, one Redis server instance can only support to store a single DocumentArray.
+```
+
+To store a new DocumentArray in current Redis server, one can set `flush` to `True` so that previous DocumentArray will be cleared:
+
+```python
+from docarray import DocumentArray
+
+da = DocumentArray(storage='redis', config={'n_dim': 128, 'flush': True})
+```
+
+To access a previously stored DocumentArray, one can specify `host` and `port`.
+
+The following example will build a DocumentArray with previously stored data on `localhost:6379`:
+
+```python
+from docarray import DocumentArray, Document
+
+with DocumentArray(
+    storage='redis',
+    config={'n_dim': 128, 'flush': True},
+) as da:
+    da.extend([Document() for _ in range(1000)])
+
+da2 = DocumentArray(
+    storage='redis',
+    config={'n_dim': 128},
+)
+
+da2.summary()
+```
+
+```text
+╭────────────── Documents Summary ──────────────╮
+│                                               │
+│   Type                   DocumentArrayRedis   │
+│   Length                 1000                 │
+│   Homogenous Documents   True                 │
+│   Common Attributes      ('id',)              │
+│   Multimodal dataclass   False                │
+│                                               │
+╰───────────────────────────────────────────────╯
+╭───────────────────── Attributes Summary ─────────────────────╮
+│                                                              │
+│   Attribute   Data type   #Unique values   Has empty value   │
+│  ──────────────────────────────────────────────────────────  │
+│   id          ('str',)    1000             False             │
+│                                                              │
+╰──────────────────────────────────────────────────────────────╯
+╭─── DocumentArrayRedis Config ───╮
+│                                 │
+│   n_dim             128         │
+│   host              localhost   │
+│   port              6379        │
+│   index_name        idx         │
+│   flush             False       │
+│   update_schema     True        │
+│   distance          COSINE      │
+│   redis_config      {}          │
+│   batch_size        64          │
+│   method            HNSW        │
+│   ef_construction   200         │
+│   m                 16          │
+│   ef_runtime        10          │
+│   block_size        1048576     │
+│   initial_cap       None        │
+│   columns           []          │
+│                                 │
+╰─────────────────────────────────╯
+```
+
+
+
+Other functions behave the same as in-memory DocumentArray.
+
+
+### Vector search with filter query
+
+One can perform Vector Similarity Search based on FLAT or HNSW algorithm and pre-filter results using a filter query that is based on [MongoDB's Query](https://www.mongodb.com/docs/manual/reference/operator/query/). We currently support a subset of those selectors:
```

Comment:
i wouldn't mention MongoDB here. BTW we can refer it as `domain specific language` in the future or [DSL](https://en.wikipedia.org/wiki/Domain-specific_language)